src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs

```diff
@@ -11,24 +12,39 @@ namespace AutoRest.Swagger.Validation
 {
     public class ProvidersPathValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
     {
-        private readonly Regex FullRegex = new Regex(@"\/Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/[^\/]+(\/[^\/]+\/\{[^\}]+\})*(\/[^\/]+)?([\?]?[^\/]+)*$", RegexOptions.IgnoreCase);
-        private readonly Regex ProviderRegex = new Regex(@"Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/.+$", RegexOptions.IgnoreCase);
+        private readonly Regex ProviderRegex = new Regex(@"Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/([^\/]+)(?<typenamesandvalues>[^?]+)([\?]?.+)?$", RegexOptions.IgnoreCase);
```

Comment:
So, even if we consider the last child type, it would still end up in even positions ```0/typevalue0/2/typevalue2/4/typevalue4/6/typevalue6/default```
I think the logic here would suffice

I shall definitely update the regex in ````JsonValidationMessages``` and move it to ```ValidationUtilities```